cluster/addons/iso-client/opaque/opaque.go

```diff
@@ -0,0 +1,142 @@
+package opaque
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/transport"
+)
+
+// struct which keeps all information necessary for advertising opaque resources
+type OpaqueIntegerResourceAdvertiser struct {
+	Node   string
+	Name   string
+	Value  string
+	Config *restclient.Config
+}
+
+// body for opaque resource request
+type opaqueIntegerResourceRequest struct {
+	Operation string `json:"op"`
+	Path      string `json:"path"`
+	Value     string `json:"value"`
+}
+
+// constructor for OpaqueIntegerResourceAdvertiser
+func NewOpaqueIntegerResourceAdvertiser(name string, value string) (*OpaqueIntegerResourceAdvertiser, error) {
+	node, err := getNode()
+	if err != nil {
+		return nil, err
+	}
+	config, err := getClientConfig()
+	if err != nil {
+		return nil, err
+	}
+	return &OpaqueIntegerResourceAdvertiser{
+		Node:   node,
+		Config: config,
+		Name:   name,
+		Value:  value,
+	}, nil
+}
+
+func (opaque *OpaqueIntegerResourceAdvertiser) toRequestBody(operation string) ([]byte, error) {
+	// body is in form of [{"op": "add", "path": "somepath","value": "somevalue"}]
+	opaqueJson, err := json.Marshal([]opaqueIntegerResourceRequest{
```

Comment:
nit: just `return json.Marhall(...)` since the error checking must be repeated at the call site.